planner/core/rule_join_reorder_dp.go

```diff
@@ -98,13 +154,13 @@ func (s *joinReorderDPSolver) bfsGraph(startNode int, visited []bool, adjacents
 	return visitID2NodeID
 }
 
-func (s *joinReorderDPSolver) dpGraph(newPos2OldPos, oldPos2NewPos []int, joinGroup []LogicalPlan, totalEdges []joinGroupEdge) (LogicalPlan, error) {
+func (s *joinReorderDPSolver) dpGraph(newPos2OldPos, oldPos2NewPos []int, joinGroup []LogicalPlan,
```

Comment:
Could you add some brief comments to introduce the DP algorithm? BTW, how about s/newPos2OldPos/visitID2NodeID/ and s/oldPos2NewPos/nodeID2VisitID/?